index_impl.go

```diff
@@ -562,6 +562,25 @@ func (i *indexImpl) SearchInContext(ctx context.Context, req *SearchRequest) (sr
 		}
 	}()
 
+	var doScoreFusion bool
+	var rescorer *rescorer
+	if _, ok := ctx.Value(search.ScoreFusionKey).(bool); !ok {
+		// Since the score fusion key is not set, check if it is a hybrid search.
+		// If it is, then set doScoreFusion. This indicates that the rescoring
+		// must happen locally within this index, and there are no other
+		// indexes/aliases involved.
+		doScoreFusion = IsScoreFusionRequired(req)
+
+		// new context will be used in internal functions to collect data
+		// as suitable for hybrid search. Rescorer is used for rescoring
+		// using fusion algorithms.
+		if doScoreFusion {
```

Comment:
Related to comment above, it seems we can simply set rescorer with `IsScoreFusionRequired(..)` and use `rescorer != nil`  for decision making everywhere else.